### PR TITLE
INC-502: Make chart templates reusable

### PR DESCRIPTION
**Pull request by ushkarev**
Now:
• There is one base chart template which adds the common wrapper including guidance and feedback boxes
• The 3 types of charts now have extensible templates: horizontal stacked percentage bars (for behaviour entries and prisoners on levels), heatmap (for prisoners with behaviour entries), vertical grouped bar charts (trends)

### CODE REVIEW

File: server/services/analyticsServiceTypes.ts [hunk 1]
```diff
@@ -63,10 +62,7 @@ export type BehaviourEntriesByLocation = {
 export type PrisonersWithEntriesByLocation = {
   location: string
   href?: string
-  prisonersWithPositive: number
-  prisonersWithNegative: number
-  prisonersWithBoth: number
-  prisonersWithNeither: number
+  values: [number, number, number, number]
```
Comment (xoen) [inline server/services/analyticsServiceTypes.ts:65]: As things are getting more generic I think they're starting to be more abstract and potentially confusing. 

In this specific case for example, the numbers had a name and you could tell what was their meaning, but now it's just an array of numbers.

I'm wondering if add some comments would be a good idea, e.g.

```
// ...
// [prisonersWithPositive, prisonersWithNegative, prisonersWithBoth, prisonersWithBoth]
values: [number, number, number, number]
```

but maybe there is a better way of retaining the semantics.

Comment (ushkarev) [reply server/services/analyticsServiceTypes.ts:65]: the reason it's all getting more generic was that the templates were going to explode soon: there will be some 30 graphs. it'd be a real pain (end error prone!) to synchronise that many templates if each one used different variables and subtle variation in logic.

i'm ok with adding extra comments tho

File: server/views/partials/analytics/chartBase.njk [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{% from "govuk/components/details/macro.njk" import govukDetails %}
+
+{#
+Required variables:
```
[the remark this answers is not in the document]
Comment (ushkarev) [reply server/views/partials/analytics/chartBase.njk:4]: conversely, it's really easy to not spot a variable that's buried deep in this template… so i dunno really 🤷

[the remark this answers is not in the document]
Comment (ushkarev) [reply server/views/partials/analytics/chartBase.njk:4]: i wonder if using a macro might work… that way the call site _must_ use the right variables and the macro template would have to define all vars on the definition line…

**Review by xoen — APPROVED** [review]

**Review by xoen — APPROVED** [review]

File: server/views/partials/analytics/chartBase.njk (hunk 2 above)
[the remark this answers is not in the document]
Comment (ushkarev) [reply server/views/partials/analytics/chartBase.njk:4]: gonna leave the conversion from extended templates into chain of macros for a different PR – the change is becoming too big and there are a couple of downsides

**Pull request merged**